### PR TITLE
API: space creation and default location

### PR DESCRIPTION
**Pull request by sevein**
This pull request brings two commits:

- API: allow space creation (connects to https://github.com/archivematica/Issues/issues/36)
- API: create default location (connects to https://github.com/archivematica/Issues/issues/37)

### CODE REVIEW

File: storage_service/locations/api/resources.py [hunk 1]
```diff
@@ -319,6 +319,11 @@ def save_m2m(self, bundle):
                     pipeline=item.obj,
                     location=bundle.obj)
 
+    def obj_create(self, bundle, **kwargs):
+        if 'default' in bundle.data:
+            kwargs['default'] = bundle.data['default']
+        return super(LocationResource, self).obj_create(bundle, **kwargs)
+
```
Comment (jrwdunham) [inline storage_service/locations/api/resources.py]: It was a bit subtle to me that `obj_create` calls `for key, value in kwargs.items(): setattr(bundle.obj, key, value)` in tastypie/resources.py, which calls the `default` attr's setter on the location, which ultimately results in `set_default_location_post_save` creating the `Setting` instance to store the default nature of the location.

**Review by jrwdunham — APPROVED** [review]
Small changes, with tests. Most excellent.

File: storage_service/locations/api/resources.py (hunk 1 above)
Comment (sevein) [reply storage_service/locations/api/resources.py]: Yeah, not great. The "default" attribute is fairly simple at the model layer but I could probably do better in `LocationResource` (tastypie). I remember trying to add `default` as a new filed in `LocationResource` hoping that'd make things much cleaner but I had some issues.

Comment (sevein) [reply storage_service/locations/api/resources.py]: I've added a comment to the method hoping that will make things easier to understand for the reader.

**Pull request merged**